### ISSUE TITLE
match_single_binding: false positive with `#[cfg]`

### ISSUE THREAD

**Issue by goffrie** [rust-lang/rust-clippy#5552]
[Sample program](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6293b542838bb18ea90a69215c8e4dcc):
```rust
fn main() {
    match 1 {
        #[cfg(windows)]
        1 => println!("windows"),
        _ => println!("otherwise"),
    }
}
```

Clippy output:
```
warning: this match could be replaced by its body itself
 --> src/main.rs:2:5
  |
2 | /     match 1 {
3 | |         #[cfg(windows)]
4 | |         1 => println!("windows"),
5 | |         _ => println!("otherwise"),
6 | |     }
  | |_____^ help: consider using the match body instead: `println!("otherwise");`
  |
  = note: `#[warn(clippy::match_single_binding)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_single_binding

warning: 1 warning emitted
```

However, following the suggestion would change behaviour on `#[cfg(windows)]`.

**Comment by flip1995** [comment]
I think this is pretty much impossible to fix, since when checking the code the `1 => ..` arm isn't even in the HIR anymore.

https://github.com/rust-lang/rust-clippy/blob/0a53ed2d8ea81d09b0ae2d6c5e206ec22a0ba46c/clippy_lints/src/matches.rs#L900-L902
